cherab/core/tests/test_line_emission.py

```diff
@@ -0,0 +1,315 @@
+# European Commission - subsequent versions of the EUPL (the "Licence");
+# You may not use this work except in compliance with the Licence.
+# You may obtain a copy of the Licence at:
+#
+# https://joinup.ec.europa.eu/software/page/eupl5
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the Licence is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied.
+#
+# See the Licence for the specific language governing permissions and limitations
+# under the Licence.
+
+import unittest
+
+import numpy as np
+
+from raysect.core import Point3D, Vector3D, translate
+from raysect.optical import World, Spectrum, Ray
+
+from cherab.core.atomic import Line, AtomicData, ImpactExcitationPEC, RecombinationPEC, ThermalCXPEC
+from cherab.core.atomic import deuterium, carbon
+from cherab.tools.plasmas.slab import build_constant_slab_plasma
+from cherab.core.model import ExcitationLine, RecombinationLine, ThermalCXLine, GaussianLine, ZeemanTriplet
+
+
+class ConstantImpactExcitationPEC(ImpactExcitationPEC):
+    """
+    Constant electron impact excitation PEC for test purpose.
+    """
+
+    def __init__(self, value):
+        self.value = value
+
+    def evaluate(self, density, temperature):
+
+        return self.value
+
+
+class ConstantRecombinationPEC(RecombinationPEC):
+    """
+    Constant recombination PEC for test purpose.
+    """
+
+    def __init__(self, value):
+        self.value = value
+
+    def evaluate(self, density, temperature):
+
+        return self.value
+
+
+class ConstantThermalCXPEC(ThermalCXPEC):
+    """
+    Constant recombination PEC for test purpose.
+    """
+
+    def __init__(self, value):
+        self.value = value
+
+    def evaluate(self, electron_density, electron_temperature, donor_temperature):
+
+        return self.value
+
+
+class TestAtomicData(AtomicData):
+    """Fake atomic data for test purpose."""
+
+    def impact_excitation_pec(self, ion, charge, transition):
+
+        return ConstantImpactExcitationPEC(1.4e-39)
+
+    def recombination_pec(self, ion, charge, transition):
+
+        return ConstantRecombinationPEC(8.e-41)
+
+    def thermal_cx_pec(self, donor_ion, donor_charge, receiver_ion, receiver_charge, transition):
+
+        return ConstantThermalCXPEC(1.2e-46)
+
+    def wavelength(self, ion, charge, transition):
+
+        return 529.27
+
+
+class TestExcitationLine(unittest.TestCase):
+
+    world = World()
+
+    atomic_data = TestAtomicData()
+
+    plasma_species = [(carbon, 5, 2.e18, 800., Vector3D(0, 0, 0))]
+    slab_length = 1.2
+    plasma = build_constant_slab_plasma(length=slab_length, width=1, height=1, electron_density=1e19, electron_temperature=1000.,
+                                        plasma_species=plasma_species, b_field=Vector3D(0, 10., 0))
+    plasma.atomic_data = atomic_data
+    plasma.parent = world
```

Comment:
Currently the tests do not change the plasma state, but you are right, it is better to move this to the `setUp()` method to avoid possible problems in the future. For all other tests I will do this in a separate PR.